transit_odp/fares_validator/tests/test_service_frame.py

```diff
@@ -0,0 +1,495 @@
+        "lines_present",
+        "line_present",
+        "line_name_present",
+        "expected",
+    ),
+    [
+        (True, True, True, True, None),
+        (False, True, True, True, None),
+        (True, False, True, True, None),
+        (
+            True,
+            True,
+            False,
+            True,
+            ["violation", "4", "Mandatory element 'Line' in ServiceFrame is missing"],
+        ),
+        (
+            True,
+            True,
+            True,
+            False,
+            [
+                "violation",
+                "5",
+                "From 'Line' element in ServiceFrame, element 'Name' is missing or empty",
+            ],
+        ),
+    ],
+)
+def test_is_lines_present_in_service_frame(
+    service_frame_present, lines_present, line_present, line_name_present, expected
+):
+    service_frame_with_child = """
+    <ServiceFrame version="1.0" id="epd:UK:FYOR:ServiceFrame_UK_PI_NETWORK:1_Inbound:op" dataSourceRef="data_source" responsibilitySetRef="tariffs">
+        <lines>
+            <Line version="1.0" id="236">
+                <Name>Southdown PSV 236</Name>
+            </Line>
+        </lines>
+    </ServiceFrame>
+    """
+    service_frame_without_line = """
+    <ServiceFrame version="1.0" id="epd:UK:FYOR:ServiceFrame_UK_PI_NETWORK:1_Inbound:op" dataSourceRef="data_source" responsibilitySetRef="tariffs">
+        <lines>
+        </lines>
+    </ServiceFrame>
+    """
+    service_frame_without_name = """
+    <ServiceFrame version="1.0" id="epd:UK:FYOR:ServiceFrame_UK_PI_NETWORK:1_Inbound:op" dataSourceRef="data_source" responsibilitySetRef="tariffs">
+        <lines>
+            <Line version="1.0" id="236">
+            </Line>
+        </lines>
+    </ServiceFrame>
+    """
+    service_frame_without_lines = """
+    <ServiceFrame version="1.0" id="epd:UK:FYOR:ServiceFrame_UK_PI_NETWORK:1_Inbound:op" dataSourceRef="data_source" responsibilitySetRef="tariffs">
+    </ServiceFrame>
+    """
+    service_frames = """<PublicationDelivery version="1.1" xsi:schemaLocation="http://www.netex.org.uk/netex http://netex.uk/netex/schema/1.09c/xsd/NeTEx_publication.xsd" xmlns="http://www.netex.org.uk/netex" xmlns:siri="http://www.siri.org.uk/siri" xmlns:gml="http://www.opengis.net/gml/3.2" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
+            {0}
+    </PublicationDelivery>"""
+    if service_frame_present:
+        if lines_present:
+            if line_present:
+                if line_name_present:
+                    xml = service_frames.format(service_frame_with_child)
+                else:
+                    xml = service_frames.format(service_frame_without_name)
+            else:
+                xml = service_frames.format(service_frame_without_line)
+        else:
+            xml = service_frames.format(service_frame_without_lines)
+    else:
+        xml = service_frames.format("")
+    service_frames = get_service_frame_lxml_element(xml)
```

Comment:
Pass xpath and xml from here to the common function

Reply:
Done